azext_edge/edge/providers/check/base/deployment.py

```diff
@@ -0,0 +1,129 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from functools import partial
+from knack.log import get_logger
+from kubernetes.client.exceptions import ApiException
+from rich.padding import Padding
+from typing import Any, Callable, Dict, List, Optional
+
+from .check_manager import CheckManager
+from .node import check_nodes
+from .resource import enumerate_ops_service_resources
+from .user_strings import UNABLE_TO_DETERMINE_VERSION_MSG
+from ..common import CoreServiceResourceKinds, ResourceOutputDetailLevel
+from ...base import client
+from ....common import CheckTaskStatus, ListableEnum
+from ....providers.edge_api import EdgeResourceApi
+
+
+logger = get_logger(__name__)
+# TODO: unit test
+
+
+def check_pre_deployment(
```

Comment:
Just a general note - we should be able to be "smart" about when this is called in the check function, based on whether the target cluster already has an AIO deployment.

If the user doesn't have an AIO deployment, we should run this check regardless of whether they specified `--pre` (unless, maybe they set `--pre false`). If they do already have AIO deployed, we can skip this check unless they specify `--pre`

Reply:
currently, it will run pre all the time. I agree with having pre run only if AIO is not installed or pre is explicitly called. 

Will change